examples/cache/Dockerfile-service

```diff
@@ -1,6 +1,6 @@
 FROM envoyproxy/envoy-alpine-dev:latest
 
-RUN apk update && apk add py3-pip bash curl
+RUN apk update && apk add py3-pip bash curl gcc python3-dev musl-dev
```

Comment:
this is unnecessary - it has been fixed upstream

Reply:
When? It was failing without it 6 hours ago.

Reply:
since then  let me find the upstream ticket ...

Reply:
https://github.com/cython/cython/issues/4119 is the upstream issue

and

https://github.com/envoyproxy/envoy/issues/15981 is the one i added to envoy

they are now both fixed - and i believe the cause of the issue you saw

Reply:
bumping just the linux image works without any other changes see #15986 

ive opened another to bump the build tools in bazel and currently waiting on ci - #15992

Reply:
See failed run: https://dev.azure.com/cncf/envoy/_build/results?buildId=72020&view=logs&j=372032e6-8bf2-5d86-e328-540457529814&t=85caf536-e2ae-5b30-3a82-f9aadd42f391

Reply:
What's the issue with adding `gcc` until upstream is fixed? There are other examples that use it.

Reply:
upstream is already fixed - why make it compile cython when there is no reason too ?

Reply:
Oh, I didn't notice the fix. I reverted this change. Thanks!